docs/index.md

```diff
@@ -27,53 +27,130 @@ can be used to:
 
 ## Instructions
 
-1.  Set up Eclipse and CLDR, according to instructions:
+### General Setup for Maven
+
+1.  Configure Maven settings (including Github tokens) according to these instructions:
     <http://cldr.unicode.org/development/maven>
+2.  Get your github account authorized for <https://github.com/unicode-org/unicodetools>,
+    create a fork under your account, and create a local clone.
+
+#### In-source / Out-of-source build setup
+
+Some of the tasks within the Unicode Tools generate output files that can also be input files into other steps.
+For this purpose, we create a folder named `Generated` to store these files.
+This folder can be a subfolder inside the local working copy root (called an "In-source build" workspace layout), or this folder can be outside (ex: a sibling folder) the local working-copy root (called an "Out-of-source build" workspace layout). Both workspace styles are described below.
+
+Out-of-source builds keep a separation between source files of the repository and their generated output files, which are no tracked in the repository. Out-of-source builds allow developers to maintain a clean view of changes to tracked source files, without mixing generated output files. (Out-of-source builds are also useful for C++ repositories in which multiple configurations can be invoked to generate independent sets of makefiles that result in corresponding different output compiled binary files.
+
+##### Setup for an out-of-source build workspace
+
+1. Create directories for cloning the Unicode Tools and CLDR repositories:
+```
+mkdir -p unicodetools/mine/src
+mkdir -p cldr/mine/src
+```
+2. Clone the repositories' contents into their respective source directories:
+```
+git clone https://github.com/unicode-org/unicodetools.git unicodetools/mine/src
+git clone https://github.com/unicode-org/cldr.git cldr/mine/src
+```
+3. Create the `Generated` folder structure as a sibling to the local working copy root:
+```
+mkdir -p unicodetools/mine/output/Generated/BIN
+ln -s ../../src/unicodetools/data/security unicodetools/mine/output/Generated/security
```

Comment:
I think that you & Steven have discussed this line in the context of CI, but in my running tools to generate security files I have so far not created such a symlink, and I am pretty sure that we used to not have any instructions for such a thing. What problem does this symlink solve?

Reply:
If I remove the symlink, the first listed command succeeds (Make Unicode Files), but the second command fails (Build and Test) because it can't find the TestSecurity classes and a related file:

```
[ERROR] Tests run: 8, Failures: 0, Errors: 8, Skipped: 0, Time elapsed: 0.554 s <<< FAILURE! - in org.unicode.test.TestSecurity
[ERROR] org.unicode.test.TestSecurity.TestCldrConsistency  Time elapsed: 0 s  <<< ERROR!
java.lang.ExceptionInInitializerError
        at org.unicode.test.TestSecurity.<clinit>(TestSecurity.java:52)
Caused by: java.io.FileNotFoundException: /usr/local/google/home/elango/oss/tmp/unicodetools-docs-testing-2/unicodetools/output/Generated/security/14.0.0/IdentifierType.txt (No such file or directory)
        at org.unicode.test.TestSecurity.<clinit>(TestSecurity.java:52)

[ERROR] org.unicode.test.TestSecurity.TestIdempotence  Time elapsed: 0.001 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.unicode.test.TestSecurity

[ERROR] org.unicode.test.TestSecurity.TestSpacing  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.unicode.test.TestSecurity

[ERROR] org.unicode.test.TestSecurity.TestXidMod  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.unicode.test.TestSecurity

[ERROR] org.unicode.test.TestSecurity.TestConfusables  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.unicode.test.TestSecurity

[ERROR] org.unicode.test.TestSecurity.TestScriptDetection  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.unicode.test.TestSecurity

[ERROR] org.unicode.test.TestSecurity.TestWholeScripts  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.unicode.test.TestSecurity

[ERROR] org.unicode.test.TestSecurity.TestWholeScriptData  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.unicode.test.TestSecurity
```

Reply:
> If I remove the symlink, the first listed command succeeds (Make Unicode Files), but the second command fails (Build and Test) because it can't find the TestSecurity classes and a related file:

That seems wrong; or rather, it seems like an artifact of trying to run TestSecurity as a unit test in isolation. I have run that test successfully, and without a symlink, according to the relevant instructions: The point is that we normally want to run TestSecurity after GenerateConfusables which writes files into .../Generated/security. In other words, we write new files and then see if they work. That's why TestSecurity wants to look at the output files.

See https://github.com/unicode-org/unicodetools/blob/main/docs/security.md

If we want to make this work as a unit test that looks at the input data without running GenerateConfusables, then TestSecurity should get a parameter for looking at input data vs. output data.

Putting an unconditional instruction for this symlink into the general setup will make GenerateConfusables malfunction.

More generally, remember that we used to not routinely run unit tests. What tests people ran they ran ad hoc after running other tools. As we move to real, continous unit tests, we need to keep in mind what the test code was intended to do, and adapt it to what we need.

Reply:
@markusicu 
> an artifact of trying to run TestSecurity as a unit test in isolation

Could TestSecurity be changed to read the `src/unicodetools/data/security` files as input instead, unless invoked by `GenerateConfusables` ? 

`security.md` could be helpful, but we kind of need the overall process description. I'm still not sure what all of the input and output terms are. I guess I could try going through each of the .md files and trying to perform their steps.

Reply:
> Could TestSecurity be changed to read the `src/unicodetools/data/security` files as input instead, unless invoked by `GenerateConfusables` ?

Sure, just a matter of programming :-)
As I said above, this could be a new parameter. Or I suppose it could automatically try the input path if there is nothing at the output path.

Let's please remove the symlink line from the developer setup instructions, get this PR in, and then fiddle with TestSecurity.

> `security.md` could be helpful, but we kind of need the overall process description. I'm still not sure what all of the input and output terms are. I guess I could try going through each of the .md files and trying to perform their steps.

We need to fix the setup instructions (this PR), and we will do a survey of all of the "tools", but it makes sense to have specific docs pages per tool or group of tools.